kong/db/schema/init.lua

```diff
@@ -882,6 +884,16 @@ function Schema:validate_field(field, value)
     return nil, validation_errors.SUBSCHEMA_ABSTRACT_FIELD
   end
 
+  if field.deprecation then
+    local old_default = field.deprecation.old_default
+    local should_warn = old_default == nil
```

Comment:
This line seems conflict with the doc in metaschema.lua "If `old_default` is set, logging is conditional to the field's value being different from the value of `old_default`.". Should it be `old_default ~= nil` or update the document like: 
"If `old_default` is not set, logging is conditional to the field's value being different from the value of `old_default`."
?

Reply:
I realize the comment wasn't very clear, I reworded it, could you check again? Thanks!

Reply:
Thanks! My previous comment also does not correctly catch the meaning of the code. But your last version makes it more clear. And sorry for the confusion comment.